.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,74 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "main" ]
+  schedule:
+    - cron: '25 13 * * 0'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
```

Comment:
We use Ubuntu 20.04 in all other places, including `charmcraft.yaml`. Don't we want to use the same version here?

Reply:
FYI we have more projects in which we run Codeql with `ubuntu-latest`, I don't say that's the right way of doing it, but just in case.

Reply:
I think @sanchezfdezjavier is right, in many other projects of ours we follow the same practice.
I don't see a contradiction between the code being analysed on `ubuntu-latest` and the charm being built on a specific ubuntu version.
Let me know your thoughts are @Gmerold  :)

Reply:
No problem. I was just asking.